File: Data Structures/LinkedList/ReverseLinkedList.cpp
class node{
	public:
		int data;
		node *next;
};
node* reverseList(node *head){
	node *pre=NULL;
	node *nxt=NULL;
	node *cur=head;
	while(cur!=NULL){
		nxt=cur->next;
		cur->next=pre;
		pre=cur;
		cur=nxt;
	}
	head=pre;
}

File: reverseLinkedlist.java
class ReverseLL
{
    // This function should reverse linked list and return
    // head of the modified linked list.
    Node reverseList(Node head)
    {
        // code here
        if(head.next==null ||  head==null )
            return head;
        Node second =head.next;
        Node start = reverseList(head.next);
        second.next=head;
        head.next=null;
        
        return start;
    }
}
